include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_kernels.h

```diff
@@ -599,35 +629,39 @@ struct __radix_sort_onesweep_kernel
         }
 
         // 2. Chained scan. Synchronization between work-groups.
-        else if (__local_tid < __bin_summary_group_size)
+        else if (__local_tid < __bin_summary_group_size && __wg_id != 0)
         {
+            // 2.1. Read the histograms scanned across work-groups
             __dpl_esimd::__ns::simd<_GlobOffsetT, __bin_width> __prev_group_hist_sum(0), __prev_group_hist;
             __dpl_esimd::__ns::simd_mask<__bin_width> __is_not_accumulated(1);
             do
             {
                 do
                 {
+                    // Read the histogram from L2, bypassing L1
+                    // L1 is assumed to be non-coherent, thus it is avoided to prevent reading stale data
                     __prev_group_hist = __dpl_esimd::__ens::lsc_block_load<
                         _GlobOffsetT, __bin_width, __dpl_esimd::__ens::lsc_data_size::default_size,
                         __dpl_esimd::__ens::cache_hint::uncached, __dpl_esimd::__ens::cache_hint::cached>(
                         __p_prev_group_hist + __local_tid * __bin_width);
-                    // Software barrier is used in order to trick the compiler,
-                    // thus it generates memory operations with better performance
+                    // Software barrier is used in order to trick the compiler
+                    // to generate memory operations in an order, which results in better performance
```

Comment:
The comma is unnecessary and I feel that we can find a better word for "trick" here
```suggestion
                    // Software barrier is used in order to motivate the compiler
                    // to generate memory operations in an order which results in better performance
```